src/main/java/linenux/command/DeleteCommand.java

```diff
@@ -0,0 +1,131 @@
+package linenux.command;
+
+import linenux.command.result.CommandResult;
+import linenux.model.Schedule;
+import linenux.model.Task;
+import linenux.util.TasksListUtil;
+
+import java.util.ArrayList;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+/**
+ * Created by yihangho on 10/4/16.
+ */
+public class DeleteCommand implements Command {
+    private static final String DELETE_PATTERN = "(?i)^delete (?<keywords>.*)$";
+    private static final String NUMBER_PATTERN = "^\\d+$";
+    private static final String CANCEL_PATTERN = "^cancel$";
+
+    private Schedule schedule;
+    private boolean requiresUserResponse;
+    private ArrayList<Task> foundTasks;
+
+    public DeleteCommand(Schedule schedule) {
+        this.schedule = schedule;
+        this.requiresUserResponse = false;
+    }
+
+    @Override
+    public boolean respondTo(String userInput) {
+        return userInput.matches(DELETE_PATTERN);
+    }
+
+    @Override
+    public CommandResult execute(String userInput) {
+        Matcher matcher = Pattern.compile(DELETE_PATTERN).matcher(userInput);
+
+        if (matcher.matches()) {
+            String keywords = matcher.group("keywords");
+            String[] keywordsArr = keywords.split("\\s+");
+            ArrayList<Task> tasks = this.schedule.search(keywordsArr);
```

Comment:
Schedule can be null. We should prob standardise the error to throw when it is null